packages/language-server/src/completion/completions.ts

```diff
@@ -261,6 +261,27 @@ function getFirstDatasourceName(lines: string[]): string | undefined {
     .trim()
 }
 
+function getFirstDatasourceProvider(lines: string[]): string | undefined {
+  // matches provider inside datasource in any position
+  // thanks to https://regex101.com for the onlie scratchpad
+  const result =
+    /datasource.*\{(\n|\N)\s*(.*\n)?\n*\s*provider\s=\s(\"(.*)\")[^}]+}/.exec(
+      lines.join('\n'),
```

Comment:
It's nitpicky, but isn't this recompiling the regex on every call and joining all the lines in the schema on every call?

Reply:
Indeed! Maybe something to optimize later, it doesn't cause issues as far as I know for now and doing it like this gives us the guarantee that the datasource didn't change in the meantime.